pkg/controller/controller_test.go

```diff
@@ -86,7 +87,13 @@ func TestHandlePopup(t *testing.T) {
 			c := New(harness.Renderer, harness.Processor, tc.allowlist)
 
 			w, r := envtest.BuildFormRequest(ctx, t, tc.method, tc.path,
-				tc.queryParam, tc.headers)
+				tc.queryParam)
+
+			for key, values := range *tc.headers {
```

Comment:
`http.Header` is a `map[string][]string`, which is already a pointer. The test cases should use `http.Header`, not `*http.Header`.

Reply:
ack i suppose the same is true for `url.Values`